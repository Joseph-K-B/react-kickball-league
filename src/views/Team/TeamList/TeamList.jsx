import {useEffect, useState} from 'react'
import { Link } from 'react-router-dom';
import { deleteTeamById, getTeams } from '../../../services/teams';

function TeamList() {
    const [loading, setLoading] = useState(true);
    const [teams, setTeams] = useState([]);

    const reloadTeams = async () => {
        setLoading(true);
        const res = await getTeams();
        setTeams(res);
        setLoading(false);
    }

    useEffect(() => {
        getTeams()
        .then((res) => setTeams(res))
        .finally(() => setLoading(false))
    }, [])

    const handleDelete = async ({ id, name }) => {
        // eslint-disable-next-line no-restricted-globals
        const confirmDelete = confirm(`Would you like to remove ${name} from this seasons roster?`);

        if(confirmDelete) {
            await deleteTeamById(id);
            await reloadTeams();
        }
    };

    if(loading) return <h1>Loading teams...</h1>
    
    return (
        <>
        <h1>Team List</h1>
        <ul>
            {teams.map((team) => {
                return (
                    <li key={team.id}>
                        <Link className='link' to={`/teams/${team.id}`}>
                        <p>{team.name}</p>
                        </Link>
                        <button
                            type="button"
                            aria-label={`Delete ${team.name}`}
                            onClick={() => handleDelete({ id: team.id, name: team.name})}
                        >
                            Delete
                        </button>
                        <Link to={`/teams/${team.id}/update`}>
                        <button
                            type="button"
                            aria-label={`Update ${team.name}`}
                        >
                            Update
                        </button>
                        </Link>
                    </li>
                    )
                })}
            </ul>            
            <Link className='link' to='/teams/new'><p>Add Team</p></Link>
            </>
    )
}

export default TeamList