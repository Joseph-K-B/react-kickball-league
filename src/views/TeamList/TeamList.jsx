import {useEffect, useState} from 'react'
import { Link } from 'react-router-dom';
import { getTeams } from '../../services/teams';

function TeamList() {
    const [loading, setLoading] = useState(true);
    const [teams, setTeams] = useState([]);

    useEffect(() => {
        getTeams()
        .then((res) => setTeams(res))
        .finally(() => setLoading(false))
    }, [])

    if(loading) return <h1>Loading teams...</h1>
    
    return (
        <>
        <h1>Team List</h1>
        <ul>
            {teams.map((team) => (
                <li key={team.id}>
                    <Link className='link' to={`/teams/${team.id}`}>
                    <p>{team.name}</p>
                    </Link>
                </li>
            ))}
        </ul>
        <Link className='link' to='/teams/new'><p>Add Team</p></Link>
        </>
    )
}

export default TeamList