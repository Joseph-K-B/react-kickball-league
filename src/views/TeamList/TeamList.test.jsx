import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import TeamList from './TeamList';

it('renders team details according to id param', async () => {
    const {container} = render(
        <MemoryRouter initialEntries={['/teams']}>
            <Route path="/teams">
                <TeamList />
            </Route>
        </MemoryRouter>
    );

    screen.getByText('Loading teams...')

    const team = await screen.findByText(/Bridge City Sneakers/, {exact: false});
    const teamTwo = await screen.findByText(/Lakeville Thunderfeet/, {exact: false});
    const teamThree = await screen.findByText(/Mt. Hoodies/, {exact: false});
    expect(team).toBeInTheDocument();
    expect(teamTwo).toBeInTheDocument();
    expect(teamThree).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})