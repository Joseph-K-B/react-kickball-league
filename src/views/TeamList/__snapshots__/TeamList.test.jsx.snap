// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team details according to id param 1`] = `
<div>
  <h1>
    Team List
  </h1>
  <ul>
    <li>
      <a
        href="/teams/10"
      >
        <p>
          Acme USA
        </p>
      </a>
    </li>
    <li>
      <a
        href="/teams/1"
      >
        <p>
          Bridge City Sneakers
        </p>
      </a>
    </li>
    <li>
      <a
        href="/teams/3"
      >
        <p>
          Lakeville Thunderfeet
        </p>
      </a>
    </li>
    <li>
      <a
        href="/teams/4"
      >
        <p>
          Mt. Hoodies
        </p>
      </a>
    </li>
    <li>
      <a
        href="/teams/2"
      >
        <p>
          Stumptown Lumberjacks
        </p>
      </a>
    </li>
    <li>
      <a
        href="/teams/5"
      >
        <p>
          Vancougers
        </p>
      </a>
    </li>
  </ul>
</div>
`;
